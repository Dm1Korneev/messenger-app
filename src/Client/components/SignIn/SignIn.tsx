import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
import PersonAddIcon from '@mui/icons-material/PersonAddOutlined';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import FormHelperText from '@mui/material/FormHelperText';
import LinearProgress from '@mui/material/LinearProgress';
import Paper from '@mui/material/Paper';
import Tab from '@mui/material/Tab';
import Tabs from '@mui/material/Tabs';
import Typography from '@mui/material/Typography';
import {
  Field, Form, Formik, FormikHelpers,
} from 'formik';
import { CheckboxWithLabel } from 'formik-material-ui';
import { useRef, useState } from 'react';
import * as Yup from 'yup';

import {
  email as emailValidation,
  name as nameValidation,
  password as passwordValidation,
} from 'Client/common/validation';
import { useLogin, useRegister } from 'Client/hooks';

import { AvatarSelector } from '../AvatarSelector';
import { UserEmailField } from '../UserEmailField';
import { UserNameField } from '../UserNameField';
import { UserPasswordField } from '../UserPasswordField';

enum DialogTabs {
  SIGN_IN = 'SIGN_IN',
  REGISTER = 'REGISTER',
}

const validationSchemaSignIn = Yup.object().shape({
  email: emailValidation,
  password: passwordValidation,
});

const validationSchemaRegister = Yup.object().shape({
  email: emailValidation,
  password: passwordValidation,
  name: nameValidation,
});

type FromValues = {
  name: string;
  email: string;
  password: string;
  remember: boolean;
}

export const SignIn = () => {
  const { mutate: login, isLoading: isLogin, error: loginError } = useLogin();
  const { mutate: register, isLoading: isRegister, error: registerError } = useRegister();

  const isLogging = isLogin || isRegister;

  const [variant, setVariant] = useState<DialogTabs>(DialogTabs.SIGN_IN);

  const avatarFileInput = useRef<HTMLInputElement>(null);

  const onSignIn = (values: FromValues) => {
    const { email, password, remember } = values;
    login({ email, password, saveToStore: remember });
  };

  const onRegister = (values: FromValues) => {
    const files = avatarFileInput?.current?.files;
    const avatar = files ? files[0] : undefined;

    const {
      email, password, name, remember,
    } = values;
    register({
      email, password, name, avatar, saveToStore: remember,
    });
  };

  const onSubmit = (values: FromValues, { setSubmitting }: FormikHelpers<FromValues>) => {
    if (variant === DialogTabs.SIGN_IN) {
      onSignIn(values);
    } else if (variant === DialogTabs.REGISTER) {
      onRegister(values);
    }
    setSubmitting(false);
  };

  let validationSchema;
  let title = '';
  if (variant === DialogTabs.SIGN_IN) {
    validationSchema = validationSchemaSignIn;
    title = 'Sign in';
  } else if (variant === DialogTabs.REGISTER) {
    validationSchema = validationSchemaRegister;
    title = 'Register';
  }

  const initialValues: FromValues = {
    name: '',
    email: '',
    password: '',
    remember: false,
  };

  return (
    <Box alignItems="center" display="flex" flexDirection="column" pt={8} px={2} width="100%">
      <Paper elevation={2}>
        <Box maxWidth="30rem">
          <AppBar position="static">
            <Tabs
              disabled={isLogging}
              onChange={(event, value) => setVariant(value)}
              textColor="inherit"
              value={variant}
              variant="fullWidth"
            >
              <Tab disabled={isLogging} label="Sign in" value={DialogTabs.SIGN_IN} />
              <Tab disabled={isLogging} label="Register" value={DialogTabs.REGISTER} />
            </Tabs>
          </AppBar>
          <Box
            alignItems="center"
            display="flex"
            flexDirection="column"
            p={3}
          >
            <Box p={1}>
              {variant === DialogTabs.SIGN_IN && <LockOutlinedIcon color="primary" fontSize="large" />}
              {variant === DialogTabs.REGISTER && <PersonAddIcon color="primary" fontSize="large" />}
            </Box>
            <Typography component="h1" variant="h5">
              {title}
            </Typography>
            <Box mt={1}>
              <Formik
                initialValues={initialValues}
                onSubmit={onSubmit}
                validationSchema={validationSchema}
              >
                {({ submitForm }) => (
                  <Form>
                    {loginError && variant === DialogTabs.SIGN_IN ? (
                      <FormHelperText error>{loginError.message}</FormHelperText>
                    ) : null}
                    {registerError && variant === DialogTabs.REGISTER ? (
                      <FormHelperText error>{registerError.message}</FormHelperText>
                    ) : null}
                    {variant === DialogTabs.REGISTER && (
                      <>
                        <AvatarSelector avatarFileInput={avatarFileInput} disabled={isLogging} />
                        <Field component={UserNameField} disabled={isLogging} name="name" />
                      </>
                    )}
                    <Field component={UserEmailField} disabled={isLogging} name="email" />
                    <Field component={UserPasswordField} disabled={isLogging} name="password" />
                    <Field
                      color="primary"
                      component={CheckboxWithLabel}
                      disabled={isLogging}
                      Label={{ label: 'Remember me' }}
                      name="remember"
                      type="checkbox"
                    />
                    {isLogging && <LinearProgress />}
                    <Box mt={3}>
                      <Button
                        color="primary"
                        disabled={isLogging}
                        fullWidth
                        onClick={submitForm}
                        type="submit"
                        variant="contained"
                      >
                        {title}
                      </Button>
                    </Box>
                  </Form>
                )}
              </Formik>
            </Box>
          </Box>
        </Box>
      </Paper>
    </Box>
  );
};
