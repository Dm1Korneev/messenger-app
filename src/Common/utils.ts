
type Filter = Parameters<typeof Array.prototype.filter>[0]
export const distinct: Filter = (value, index, self) => self.indexOf(value) === index;
