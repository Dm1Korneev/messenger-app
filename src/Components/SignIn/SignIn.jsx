import React, { useRef, useState } from 'react';
import PropTypes from 'prop-types';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import FormHelperText from '@material-ui/core/FormHelperText';
import LinearProgress from '@material-ui/core/LinearProgress';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import PersonAddIcon from '@material-ui/icons/PersonAddOutlined';
import { Field, Form, Formik } from 'formik';
import { CheckboxWithLabel } from 'formik-material-ui';
import * as Yup from 'yup';

import AvatarSelector from 'Components/AvatarSelector';
import UserEmailField from 'Components/UserEmailField';
import UserNameField from 'Components/UserNameField';
import UserPasswordField from 'Components/UserPasswordField';
import {
  email as emailValidation,
  name as nameValidation,
  password as passwordValidation,
} from 'Common/validation';

const SIGN_IN = 'SIGN_IN';
const REGISTER = 'REGISTER';

const validationSchemaSignIn = Yup.object().shape({
  email: emailValidation,
  password: passwordValidation,
});

const validationSchemaRegister = Yup.object().shape({
  email: emailValidation,
  password: passwordValidation,
  name: nameValidation,
});

const SignIn = ({
  loginError, registerError, onSignIn, onRegister, isLoging,
}) => {
  const [variant, setVariant] = useState(SIGN_IN);

  const avatarFileInput = useRef();

  const signIn = (values) => {
    const { email, password, remember } = values;
    onSignIn({ email, password, remember });
  };

  const register = (values) => {
    const avatar = avatarFileInput.current.files[0];

    const {
      email, password, name, remember,
    } = values;
    onRegister({
      email, password, name, avatar, remember,
    });
  };

  const onSubmit = (values, { setSubmitting }) => {
    if (variant === SIGN_IN) {
      signIn(values);
    } else if (variant === REGISTER) {
      register(values);
    }
    setSubmitting(false);
  };

  const handleTabChange = (event, value) => {
    setVariant(value);
  };

  let validationSchema;
  let title;
  if (variant === SIGN_IN) {
    validationSchema = validationSchemaSignIn;
    title = 'Sign in';
  } else if (variant === REGISTER) {
    validationSchema = validationSchemaRegister;
    title = 'Register';
  }

  return (
    <Box pt={8} px={2} width="100%" alignItems="center" display="flex" flexDirection="column">
      <Paper elevation={2}>
        <Box maxWidth="30rem">
          <AppBar position="static">
            <Tabs
              value={variant}
              onChange={handleTabChange}
              variant="fullWidth"
              disabled={isLoging}
            >
              <Tab disabled={isLoging} value={SIGN_IN} label="Sign in" />
              <Tab disabled={isLoging} value={REGISTER} label="Register" />
            </Tabs>
          </AppBar>
          <Box
            display="flex"
            flexDirection="column"
            alignItems="center"
            p={3}
          >
            <Box p={1}>
              {variant === SIGN_IN && <LockOutlinedIcon color="primary" fontSize="large" />}
              {variant === REGISTER && <PersonAddIcon color="primary" fontSize="large" />}
            </Box>
            <Typography component="h1" variant="h5">
              {title}
            </Typography>
            <Box mt={1}>
              <Formik
                initialValues={{
                  name: '',
                  email: '',
                  password: '',
                  remember: false,
                }}
                validationSchema={validationSchema}
                onSubmit={onSubmit}
              >
                {({ submitForm }) => (
                  <Form>
                    {loginError && variant === SIGN_IN && (
                    <FormHelperText error>{loginError}</FormHelperText>
                    )}
                    {registerError && variant === REGISTER && (
                    <FormHelperText error>{registerError}</FormHelperText>
                    )}
                    {variant === REGISTER && (
                    <>
                      <AvatarSelector disabled={isLoging} avatarFileInput={avatarFileInput} />
                      <Field disabled={isLoging} component={UserNameField} name="name" />
                    </>
                    )}
                    <Field disabled={isLoging} component={UserEmailField} name="email" />
                    <Field disabled={isLoging} component={UserPasswordField} name="password" />
                    <Field
                      disabled={isLoging}
                      component={CheckboxWithLabel}
                      name="remember"
                      Label={{ label: 'Remember me' }}
                      color="primary"
                      type="checkbox"
                    />
                    {isLoging && <LinearProgress />}
                    <Box mt={3}>
                      <Button
                        type="submit"
                        fullWidth
                        variant="contained"
                        color="primary"
                        disabled={isLoging}
                        onClick={submitForm}
                      >
                        {title}
                      </Button>
                    </Box>
                  </Form>
                )}
              </Formik>
            </Box>
          </Box>
        </Box>
      </Paper>
    </Box>
  );
};

SignIn.defaultProps = {
  loginError: null,
  registerError: null,
  isLoging: false,
};
SignIn.propTypes = {
  onSignIn: PropTypes.func.isRequired,
  onRegister: PropTypes.func.isRequired,
  isLoging: PropTypes.bool,
  loginError: PropTypes.string,
  registerError: PropTypes.string,
};

export default SignIn;
