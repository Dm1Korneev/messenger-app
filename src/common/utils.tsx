export function distinct(value: any, index: number, self: Array<any>): boolean {
  return self.indexOf(value) === index;
}
