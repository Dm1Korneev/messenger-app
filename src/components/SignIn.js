import React from "react";
import { ValidatorForm } from "react-material-ui-form-validator";

// material-ui
import { withStyles } from "@material-ui/core/styles";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import AppBar from "@material-ui/core/AppBar";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import FormHelperText from "@material-ui/core/FormHelperText";

// icons
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import PersonAddIcon from "@material-ui/icons/PersonAddOutlined";

import AvatarSelector from "./AvatarSelector";
import UserPasswordField from "./UserPasswordField";
import UserNameField from "./UserNameField";
import UserEmailField from "./UserEmailField";

const SIGN_IN = "SIGN_IN";
const REGISTER = "REGISTER";

class SignIn extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
      name: "",
      remember: false,
      errorMessage: undefined,
      avatar: undefined,
      variant: SIGN_IN
    };
    this.avatarFileInput = React.createRef();
  }

  handleSubmit = () => {
    const { variant } = this.state;

    if (variant === SIGN_IN) {
      this.signIn();
    } else if (variant === REGISTER) {
      this.register();
    }
  };

  signIn = () => {
    const { onSignIn } = this.props;
    const { email, password, remember } = this.state;
    onSignIn(email, password, remember);
  };

  register = () => {
    const avatar = this.avatarFileInput.current.files[0];

    const { onRegister } = this.props;
    const { email, password, name, remember } = this.state;
    onRegister(email, password, name, avatar, remember);
  };

  handleInputChange = event => {
    const target = event.target;
    const value = target.type === "checkbox" ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  };

  handleTabChange = (event, value) => {
    this.setState({ variant: value });
  };

  render() {
    const { loginError, registerError, classes } = this.props;
    const { email, password, name, remember, variant } = this.state;

    return (
      <main className={classes.main}>
        <CssBaseline />
        <Paper className={classes.paper}>
          <AppBar position="static">
            <Tabs
              value={variant}
              onChange={this.handleTabChange}
              variant="fullWidth"
              classes={{ indicator: classes.tabsIndicator }}
            >
              <Tab value={SIGN_IN} label="Sign in" />
              <Tab value={REGISTER} label="Register" />
            </Tabs>
          </AppBar>
          <div className={classes.container}>
            <Avatar className={classes.icon}>
              {variant === SIGN_IN && <LockOutlinedIcon />}
              {variant === REGISTER && <PersonAddIcon />}
            </Avatar>
            <Typography component="h1" variant="h5">
              {variant === SIGN_IN && "Sign in"}
              {variant === REGISTER && "Register"}
            </Typography>
            <ValidatorForm
              className={classes.form}
              onSubmit={this.handleSubmit}
            >
              {loginError && variant === SIGN_IN && (
                <FormHelperText error>{loginError}</FormHelperText>
              )}
              {registerError && variant === REGISTER && (
                <FormHelperText error>{registerError}</FormHelperText>
              )}
              {variant === REGISTER && (
                <>
                  <AvatarSelector avatarFileInput={this.avatarFileInput} />
                  <UserNameField
                    value={name}
                    onChange={this.handleInputChange}
                  />
                </>
              )}
              <UserEmailField value={email} onChange={this.handleInputChange} />
              <UserPasswordField
                value={password}
                onChange={this.handleInputChange}
              />
              <FormControlLabel
                control={
                  <Checkbox
                    value="remember"
                    name="remember"
                    checked={remember}
                    color="primary"
                    onChange={this.handleInputChange}
                  />
                }
                label="Remember me"
              />
              <Button
                type="submit"
                fullWidth
                variant="contained"
                color="primary"
                className={classes.submit}
              >
                {variant === SIGN_IN && "Sign in"}
                {variant === REGISTER && "Register"}
              </Button>
            </ValidatorForm>
          </div>
        </Paper>
      </main>
    );
  }
}

const styles = theme => ({
  main: {
    width: "auto",
    display: "block", // Fix IE 11 issue.
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(500 + theme.spacing.unit * 3 * 2)]: {
      width: 500,
      marginLeft: "auto",
      marginRight: "auto"
    }
  },
  container: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme
      .spacing.unit * 3}px`
  },
  paper: {
    marginTop: theme.spacing.unit * 8
  },
  icon: {
    margin: theme.spacing.unit,
    backgroundColor: theme.palette.primary.main
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing.unit
  },
  submit: {
    marginTop: theme.spacing.unit * 3
  },
  tabsIndicator: {
    backgroundColor: "#fff"
  }
});

export default withStyles(styles)(SignIn);
