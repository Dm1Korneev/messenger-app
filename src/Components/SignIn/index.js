import commonHoc from 'Components/commonHoc';
import { LOGIN, REGISTER } from 'Constants/actionNames';
import { register, signIn } from 'Redux/actions';
import { errorSelector } from 'Selectors/errors';
import { loadingSelector } from 'Selectors/loading';

import SignIn from './SignIn';

const mapStateToProps = (state) => ({
  loginError: errorSelector(state, LOGIN),
  registerError: errorSelector(state, REGISTER),
  isLoging: loadingSelector(state, LOGIN) || loadingSelector(state, REGISTER),
});

const mapDispatchToProps = {
  onSignIn: signIn,
  onRegister: register,
};

export default commonHoc(SignIn, {
  mapStateToProps,
  mapDispatchToProps,
});

