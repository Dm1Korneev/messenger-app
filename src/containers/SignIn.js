import { connect } from "react-redux";
import { signIn, register } from "../redux/actions";
import SignIn from "../components/SignIn";

const mapStateToProps = state => ({});

const mapDispatchToProps = dispatch => ({
  onSignIn: (email, password, remember) =>
    dispatch(signIn(email, password, remember)),
  onRegister: (email, password, name, remember) =>
    dispatch(register(email, password, name, remember))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignIn);
