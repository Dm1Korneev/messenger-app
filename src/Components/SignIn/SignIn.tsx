import AppBar from '@material-ui/core/AppBar';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import FormHelperText from '@material-ui/core/FormHelperText';
import LinearProgress from '@material-ui/core/LinearProgress';
import Paper from '@material-ui/core/Paper';
import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import Typography from '@material-ui/core/Typography';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import PersonAddIcon from '@material-ui/icons/PersonAddOutlined';
import {
  Field, Form, Formik, FormikHelpers,
} from 'formik';
import { CheckboxWithLabel } from 'formik-material-ui';
import { useRef, useState } from 'react';
import { useDispatch } from 'react-redux';
import * as Yup from 'yup';

import {
  email as emailValidation,
  name as nameValidation,
  password as passwordValidation,
} from 'Common/validation';
import { AvatarSelector } from 'Components/AvatarSelector';
import { UserEmailField } from 'Components/UserEmailField';
import { UserNameField } from 'Components/UserNameField';
import { UserPasswordField } from 'Components/UserPasswordField';
import { useLogin, useRegister } from 'Hooks';
import { signIn } from 'Redux/actions';

enum DialogTabs {
  SIGN_IN = 'SIGN_IN',
  REGISTER = 'REGISTER',
}

const validationSchemaSignIn = Yup.object().shape({
  email: emailValidation,
  password: passwordValidation,
});

const validationSchemaRegister = Yup.object().shape({
  email: emailValidation,
  password: passwordValidation,
  name: nameValidation,
});

type FromValues = {
  name: string;
  email: string;
  password: string;
  remember: boolean;
}

export const SignIn = () => {
  const dispatch = useDispatch();

  const { mutate: login, isLoading: isLogin, error: loginError } = useLogin();
  const { mutate: register, isLoading: isRegister, error: registerError } = useRegister();

  const isLogging = isLogin || isRegister;

  const [variant, setVariant] = useState<DialogTabs>(DialogTabs.SIGN_IN);

  const avatarFileInput = useRef<HTMLInputElement>(null);

  const onSignIn = (values: FromValues) => {
    const { email, password, remember } = values;
    dispatch(signIn({ email, password, remember }));
    login({ email, password, saveToStore: remember });
  };

  const onRegister = (values: FromValues) => {
    const files = avatarFileInput?.current?.files;
    const avatar = files ? files[0] : undefined;

    const {
      email, password, name, remember,
    } = values;
    register({
      email, password, name, avatar, saveToStore: remember,
    });
  };

  const onSubmit = (values: FromValues, { setSubmitting }: FormikHelpers<FromValues>) => {
    if (variant === DialogTabs.SIGN_IN) {
      onSignIn(values);
    } else if (variant === DialogTabs.REGISTER) {
      onRegister(values);
    }
    setSubmitting(false);
  };

  let validationSchema;
  let title = '';
  if (variant === DialogTabs.SIGN_IN) {
    validationSchema = validationSchemaSignIn;
    title = 'Sign in';
  } else if (variant === DialogTabs.REGISTER) {
    validationSchema = validationSchemaRegister;
    title = 'Register';
  }

  const initialValues: FromValues = {
    name: '',
    email: '',
    password: '',
    remember: false,
  };

  return (
    <Box pt={8} px={2} width="100%" alignItems="center" display="flex" flexDirection="column">
      <Paper elevation={2}>
        <Box maxWidth="30rem">
          <AppBar position="static">
            <Tabs
              value={variant}
              onChange={(event, value) => setVariant(value)}
              variant="fullWidth"
              disabled={isLogging}
            >
              <Tab disabled={isLogging} value={DialogTabs.SIGN_IN} label="Sign in" />
              <Tab disabled={isLogging} value={DialogTabs.REGISTER} label="Register" />
            </Tabs>
          </AppBar>
          <Box
            display="flex"
            flexDirection="column"
            alignItems="center"
            p={3}
          >
            <Box p={1}>
              {variant === DialogTabs.SIGN_IN && <LockOutlinedIcon color="primary" fontSize="large" />}
              {variant === DialogTabs.REGISTER && <PersonAddIcon color="primary" fontSize="large" />}
            </Box>
            <Typography component="h1" variant="h5">
              {title}
            </Typography>
            <Box mt={1}>
              <Formik
                initialValues={initialValues}
                validationSchema={validationSchema}
                onSubmit={onSubmit}
              >
                {({ submitForm }) => (
                  <Form>
                    {loginError && variant === DialogTabs.SIGN_IN && (
                      <FormHelperText error>{String(loginError)}</FormHelperText>
                    )}
                    {registerError && variant === DialogTabs.REGISTER && (
                      <FormHelperText error>{String(registerError)}</FormHelperText>
                    )}
                    {variant === DialogTabs.REGISTER && (
                      <>
                        <AvatarSelector disabled={isLogging} avatarFileInput={avatarFileInput} />
                        <Field disabled={isLogging} component={UserNameField} name="name" />
                      </>
                    )}
                    <Field disabled={isLogging} component={UserEmailField} name="email" />
                    <Field disabled={isLogging} component={UserPasswordField} name="password" />
                    <Field
                      disabled={isLogging}
                      component={CheckboxWithLabel}
                      name="remember"
                      Label={{ label: 'Remember me' }}
                      color="primary"
                      type="checkbox"
                    />
                    {isLogging && <LinearProgress />}
                    <Box mt={3}>
                      <Button
                        type="submit"
                        fullWidth
                        variant="contained"
                        color="primary"
                        disabled={isLogging}
                        onClick={submitForm}
                      >
                        {title}
                      </Button>
                    </Box>
                  </Form>
                )}
              </Formik>
            </Box>
          </Box>
        </Box>
      </Paper>
    </Box>
  );
};
