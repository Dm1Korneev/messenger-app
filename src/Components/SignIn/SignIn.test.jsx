import React from 'react';
import { render } from '@testing-library/react';

import { mockComponent } from 'testing/utils';

import SignIn from './SignIn';

jest.mock('Components/UserEmailField', () => mockComponent('UserEmailField'));
jest.mock('Components/UserNameField', () => mockComponent('UserNameField'));
jest.mock('Components/UserPasswordField', () => mockComponent('UserPasswordField'));
jest.mock('Components/AvatarSelector', () => mockComponent('AvatarSelector'));

jest.mock('@material-ui/core/Avatar', () => mockComponent('Avatar'));
jest.mock('@material-ui/core/Button', () => mockComponent('Button'));
jest.mock('@material-ui/core/CssBaseline', () => mockComponent('CssBaseline'));
jest.mock('@material-ui/core/FormControlLabel', () => mockComponent('FormControlLabel'));
jest.mock('@material-ui/core/Checkbox', () => mockComponent('Checkbox'));
jest.mock('@material-ui/core/Paper', () => mockComponent('Paper'));
jest.mock('@material-ui/core/Typography', () => mockComponent('Typography'));
jest.mock('@material-ui/core/AppBar', () => mockComponent('AppBar'));
jest.mock('@material-ui/core/Tabs', () => mockComponent('Tabs'));
jest.mock('@material-ui/core/Tab', () => mockComponent('Tab'));
jest.mock('@material-ui/core/FormHelperText', () => mockComponent('FormHelperText'));
jest.mock('@material-ui/core/Box', () => mockComponent('Box'));
jest.mock('@material-ui/icons/LockOutlined', () => mockComponent('LockOutlined'));
jest.mock('@material-ui/icons/PersonAddOutlined', () => mockComponent('PersonAddOutlined'));

const props = {
  onSignIn: jest.fn(),
  onRegister: jest.fn(),
};

describe('render SignIn component', () => {
  const setup = () => render(
    <SignIn {...props} />,
  );

  test('snapshot test', async () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});
