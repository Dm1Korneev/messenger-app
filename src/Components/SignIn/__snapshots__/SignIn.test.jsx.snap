// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SignIn component snapshot test 1`] = `
<DocumentFragment>
  <main
    class="SignIn-main-1"
  >
    <div
      data-originalcomponent="CssBaseline"
    />
    <div
      data-originalcomponent="Paper"
      test-prop-classname="\\"SignIn-paper-3\\""
    >
      <div
        data-originalcomponent="AppBar"
        test-prop-position="\\"static\\""
      >
        <div
          data-originalcomponent="Tabs"
          test-prop-classes="{\\"indicator\\":\\"SignIn-tabsIndicator-7\\"}"
          test-prop-value="\\"SIGN_IN\\""
          test-prop-variant="\\"fullWidth\\""
        >
          <div
            data-originalcomponent="Tab"
            test-prop-label="\\"Sign in\\""
            test-prop-value="\\"SIGN_IN\\""
          />
          <div
            data-originalcomponent="Tab"
            test-prop-label="\\"Register\\""
            test-prop-value="\\"REGISTER\\""
          />
        </div>
      </div>
      <div
        class="SignIn-container-2"
      >
        <div
          data-originalcomponent="Avatar"
          test-prop-classname="\\"SignIn-icon-4\\""
        >
          <div
            data-originalcomponent="LockOutlined"
          />
        </div>
        <div
          data-originalcomponent="Typography"
          test-prop-component="\\"h1\\""
          test-prop-variant="\\"h5\\""
        >
          Sign in
        </div>
        <form
          class="SignIn-form-5"
        >
          <div
            data-originalcomponent="UserEmailField"
            test-prop-value="\\"\\""
          />
          <div
            data-originalcomponent="UserPasswordField"
            test-prop-value="\\"\\""
          />
          <div
            data-originalcomponent="FormControlLabel"
            test-prop-control="<div data-originalcomponent=\\"Checkbox\\" test-prop-value=\\"&quot;remember&quot;\\" test-prop-name=\\"&quot;remember&quot;\\" test-prop-checked=\\"false\\" test-prop-color=\\"&quot;primary&quot;\\" data-reactroot=\\"\\"></div>"
            test-prop-label="\\"Remember me\\""
          />
          <div
            data-originalcomponent="Button"
            test-prop-classname="\\"SignIn-submit-6\\""
            test-prop-color="\\"primary\\""
            test-prop-fullwidth="true"
            test-prop-type="\\"submit\\""
            test-prop-variant="\\"contained\\""
          >
            Sign in
          </div>
        </form>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
